cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -200,5 +211,48 @@ private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
     }
     return filePath;
   }
+
+  @Override
+  public ListAppsResponse list(NamespaceId namespace, RepositoryConfig repoConfig) throws
+    AuthenticationConfigException, NotFoundException {
+    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+      String currentCommit = repositoryManager.cloneRemote();
+      Path basePath = repositoryManager.getBasePath();
+
+      if (!Files.exists(basePath)) {
+        throw new NotFoundException(String.format("Missing repository base path %s", basePath));
+      }
+
+      FileFilter configFileFilter = getConfigFileFilter();
+      List<File> configFiles = DirUtils.listFiles(basePath.toFile(), configFileFilter);
+
+      List<ListAppResponse> responses = new ArrayList<>();
+      for (File configFile : configFiles) {
+        String applicationName = FileUtils.getNameWithoutExtension(configFile.getName());
+        String fileHash = repositoryManager.getFileHash(basePath.relativize(configFile.toPath()), currentCommit);
```

Comment:
it seems like there should be a better way to do this than manually one by one. Does jgit have the equivalent of `git ls-files -s`?

Reply:
We can walk the git tree under the basepath in the same function rather than doing per file. 
In my testing it did not provide significant time difference hence kept it simple and inline with existing `getFileHash` function. 
I can add a TODO and take this up as a fast followup from this PR.